### PR TITLE
Add support for hostname request in lwIP interface

### PR DESCRIPTION
**Pull request by guilhermerc**
### Summary of changes 

Add support for hostname request in lwIP interface.

There is a discussion about this functionality here: [How to set device network hostname on MBED OS 6?](https://forums.mbed.com/t/how-to-set-device-network-hostname-on-mbed-os-6/10086/8)

### Documentation 

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [X] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results 

    [] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [X] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by guilhermerc** [comment]
I submitted this PR to ARM's repository right before Mbed OS EOL: https://github.com/ARMmbed/mbed-os/pull/15506. Me and @multiplemonomials already had some discussion there.

**Review by multiplemonomials — COMMENTED** [review]
Looks good!  Just one comment

File: connectivity/netsocket/include/netsocket/NetworkInterface.h [hunk 1]
```diff
@@ -90,6 +90,22 @@ class NetworkInterface: public DNS {
...
+     */
+    virtual const char *get_hostname();
+
+    /** Set hostname.
+     *
+     *  @param          hostname Hostname string
+     *  @retval         NSAPI_ERROR_OK on success
+     *  @retval         NSAPI_ERROR_UNSUPPORTED if this feature is not supported
+     *  @retval         NSAPI_ERROR_PARAMETER if hostname is not valid
+     *  @retval         NSAPI_ERROR_BUSY if hostname can't be set
```
[the remark this answers is not in the document]
Comment (guilhermerc) [reply connectivity/netsocket/include/netsocket/NetworkInterface.h]: Right! Just did it.

**Review by multiplemonomials — APPROVED** [review]

**Comment by guilhermerc** [comment]
Fixed CI complaints.

**Pull request merged**